app/src/main/java/hu/vmiklos/plees_tracker/MainActivity.kt

```diff
@@ -313,7 +315,20 @@ class MainActivity : AppCompatActivity(), View.OnClickListener {
         val startStop = findViewById<FloatingActionButton>(R.id.start_stop)
         val startStopText = findViewById<TextView>(R.id.start_stop_text)
 
+        val dndManager = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        val preferences = PreferenceManager.getDefaultSharedPreferences(applicationContext)
+        val dndEnabled = preferences.getBoolean("enable_dnd", false)
+
         if (DataModel.start != null && DataModel.stop != null) {
+            // When user stops tracking sleep
+            if (dndManager.isNotificationPolicyAccessGranted and dndEnabled) {
+                // Restore Do Not Disturb status when user started tracking
+                val filterAll = NotificationManager.INTERRUPTION_FILTER_ALL
+                dndManager.setInterruptionFilter(preferences.getInt("current_dnd", filterAll))
+            } else {
+                Log.w(TAG, "Failed to disable DND, permissions not enabled")
```

Comment:
Would be ideal to only warn when this is enabled but the access is not granted.

Reply:
True, line 324 used to be `if (dndManager.isNotificationPolicyAccessGranted)` before I added the ability to enable/disable the feature in the settings which is why it was like that.